sql/sql_acl.cc

```diff
@@ -14340,7 +14341,27 @@ bool acl_authenticate(THD *thd, uint com_change_user_pkt_len)
   if (res == CR_OK_HANDSHAKE_COMPLETE)
     thd->get_stmt_da()->disable_status();
   else
-    my_ok(thd);
+  {
+    char *msg = NULL;
+
+    if (redirect_enabled)
+    {
+      char msg_buf[MAX_REDIRECTION_LEN];
```

Comment:
please declare msg_buf in the same scope where msg ist declared,, i.e abve the "if"
Otherwise, you'll potentially setting msg to a variable that is  out-of-scope. when used.

Reply:
Got it, made the change.